### PR TITLE
Backports 0.15 pr18

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: src/test/sighash_tests.cpp [hunk 1]
```diff
@@ -90,30 +89,30 @@ uint256 static SignatureHashOld(CScript scriptCode, const CTransaction& txTo, un
...
-    int ops = (insecure_rand() % 10);
+    int ops = (InsecureRandRange(10));
     for (int i=0; i<ops; i++)
-        script << oplist[insecure_rand() % (sizeof(oplist)/sizeof(oplist[0]))];
+        script << oplist[InsecureRandRange(sizeof(oplist)/sizeof(oplist[0]))];
 }
 
 void static RandomTransaction(CMutableTransaction &tx, bool fSingle) {
-    tx.nVersion = (insecure_rand() % 2) + 1;
+    tx.nVersion = (InsecureRandBool()) + 1;
```
Comment (codablock) [inline src/test/sighash_tests.cpp]: This doesn't look correct. The intention here was not to get a random bool, but either a 1 or a 2, so the correct thing would now be `InsecureRandRange(2) + 1`

File: src/test/coins_tests.cpp [hunk 2]
```diff
@@ -145,8 +145,8 @@ BOOST_AUTO_TEST_CASE(coins_cache_simulation_test)
             // Determine whether to test HaveCoin before or after Access* (or both). As these functions
             // can influence each other's behaviour by pulling things into the cache, all combinations
             // are tested.
-            bool test_havecoin_before = (insecure_rand() & 0x3) == 0; // TODO change to InsecureRandBits(2) when backporting Bitcoin #10321
-            bool test_havecoin_after = (insecure_rand() & 0x3) == 0; // TODO change to InsecureRandBits(2) when backporting Bitcoin #10321
+            bool test_havecoin_before = (InsecureRandBits(2)) == 0;
```
Comment (codablock) [inline src/test/coins_tests.cpp]: Why the unnecessary ()?

File: src/rpc/rawtransaction.cpp [hunk 3]
```diff
@@ -906,6 +906,7 @@ UniValue sendrawtransaction(const JSONRPCRequest& request)
         }
         CValidationState state;
         bool fMissingInputs;
+        bool fLimitFree = true;
```
Comment (codablock) [inline src/rpc/rawtransaction.cpp]: This one is used. I slightly remember that this variable was somehow related to a vulnerability for which we backported a fix, and this fix removed that variable completely.

File: src/util.cpp [hunk 4]
```diff
@@ -81,10 +81,12 @@
 #endif
 
 #include <boost/algorithm/string/case_conv.hpp> // for to_lower()
+#include <boost/algorithm/string/join.hpp>
```
Comment (codablock) [inline src/util.cpp]: The commit message doesn't match with what happens here...you probably mean "add back used includes" :)
What about just squashing this into the previous commit?

File: src/rpc/rawtransaction.cpp (hunk 3 above)
Comment (codablock) [reply src/rpc/rawtransaction.cpp]: I meant "is **un**used"

File: src/util.cpp (hunk 4 above)
Comment (PastaPastaPasta) [reply src/util.cpp]: done

File: src/test/coins_tests.cpp (hunk 2 above)
Comment (PastaPastaPasta) [reply src/test/coins_tests.cpp]: https://github.com/dashpay/dash/pull/3021/commits/8c6f51ee19464d62e282246a8e3195aaefd06d39

File: src/test/sighash_tests.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/test/sighash_tests.cpp]: https://github.com/dashpay/dash/pull/3021/commits/8c6f51ee19464d62e282246a8e3195aaefd06d39

File: src/test/coins_tests.cpp (hunk 2 above)
Comment (PastaPastaPasta) [reply src/test/coins_tests.cpp]: https://github.com/dashpay/dash/pull/3021/commits/84b288ea7a3a00d9f4928a3ce29af8fef32f2117

File: src/rpc/rawtransaction.cpp (hunk 3 above)
Comment (PastaPastaPasta) [reply src/rpc/rawtransaction.cpp]: removed in latest force push

**Pull request merged**